qutebrowser/config/configfiles.py

```diff
@@ -232,8 +254,17 @@ def read_config_py(filename=None):
             "Unhandled exception",
             exception=e, traceback=traceback.format_exc()))
 
-    api.finalize()
-    return api
+
+def _pre_config_save():
+    old_path = sys.path
+    old_modules = sys.modules.copy()
+    return (old_path, old_modules)
+
+
+def _post_config_load(save_tuple):
+    sys.path = save_tuple[0]
+    for module in set(sys.modules).difference(save_tuple[1]):
+        del sys.modules[module]
```

Comment:
Do you do this like this because re-assigning `sys.modules` doesn't work? If so, you might get around it by doing `sys.path[:] = saved_sys_path`, which mutates the existing list.

Reply:
Yup, re-assigning sys.modules didn't seem to work.

sys.path is a dictionary, so that method dosen't work (`TypeError: unhashable type: 'slice'`).

If you have other ideas, they would be nice, because this is a hack...